### ISSUE TITLE
Error with inspecting file with Lint/RequireRelativeSelfPath

### ISSUE THREAD

**Issue by catacs** [rubocop/rubocop#10162]
New rule `Lint/RequireRelativeSelfPath: # new in 1.22` is generating an error if the inspected file contains a call to `require_relative` with a variable.

Examples:

Example 1:
```ruby
# frozen_string_literal: true

file = "string"
require_relative(file)
```

Example 2:
```ruby
# frozen_string_literal: true

files = Dir["#{Rails.root}/doc/**/*.rb"]
files.each { |f| require_relative(f) }
```
--------

## Expected behaviour

When `require_relative` is using a variable to load file, rubocop should ignore it or at least if the string is easily inferred (direct assignment) should be smart enough to check the required file is not the same.

## Actual behaviour

When `require_relative` is called with variable generates an error: `An error occurred while Lint/RequireRelativeSelfPath cop was inspecting ....`

## Steps to reproduce the problem

Create a file like this one:

```ruby
# test.rb

# frozen_string_literal: true

file = "string"
require_relative(file)
```
running `rubocop` generates an error:

```
rubocop test.rb --debug
For $HOME/test: configuration from $HOME/test/.rubocop.yml
Inheriting configuration from $HOME/test/.rubocop_todo.yml
configuration from $HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-rspec-2.5.0/config/default.yml
configuration from $HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-rspec-2.5.0/config/default.yml
Default configuration from $HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/config/default.yml
Use parallel by default.
Running parallel inspection
An error occurred while Lint/RequireRelativeSelfPath cop was inspecting $HOME/test/test.rb:4:0.
undefined method `value' for s(:lvar, :file):RuboCop::AST::Node
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/lint/require_relative_self_path.rb:30:in `on_send'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:136:in `public_send'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:134:in `each'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-ast-1.12.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-ast-1.12.0/lib/rubocop/ast/traversal.rb:137:in `each'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-ast-1.12.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-ast-1.12.0/lib/rubocop/ast/traversal.rb:20:in `walk'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/team.rb:83:in `investigate'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:309:in `inspect_file'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:253:in `block in do_inspection_loop'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:287:in `block in iterate_until_no_changes'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:280:in `loop'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:280:in `iterate_until_no_changes'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:249:in `do_inspection_loop'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:130:in `block in file_offenses'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:155:in `file_offense_cache'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:129:in `file_offenses'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:67:in `block in warm_cache'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:515:in `call_with_index'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:485:in `process_incoming_jobs'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:465:in `block in worker'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:456:in `fork'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:456:in `worker'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:447:in `block in create_workers'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:446:in `each'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:446:in `each_with_index'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:446:in `create_workers'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:386:in `work_in_processes'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:289:in `map'
$HOME/.rvm/gems/ruby-2.7.4/gems/parallel-1.21.0/lib/parallel.rb:238:in `each'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:67:in `warm_cache'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:46:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command.rb:11:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/environment.rb:18:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli.rb:71:in `run_command'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli.rb:78:in `execute_runners'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli.rb:47:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/exe/rubocop:12:in `block in <top (required)>'
$HOME/.rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/exe/rubocop:12:in `<top (required)>'
$HOME/.rvm/gems/ruby-2.7.4/bin/rubocop:25:in `load'
$HOME/.rvm/gems/ruby-2.7.4/bin/rubocop:25:in `<main>'
$HOME/.rvm/gems/ruby-2.7.4/bin/ruby_executable_hooks:22:in `eval'
$HOME/.rvm/gems/ruby-2.7.4/bin/ruby_executable_hooks:22:in `<main>'
Inspecting 1 file
Scanning $HOME/test/test.rb
An error occurred while Lint/RequireRelativeSelfPath cop was inspecting $HOME/test/test.rb:4:0.
undefined method `value' for s(:lvar, :file):RuboCop::AST::Node
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/lint/require_relative_self_path.rb:30:in `on_send'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:136:in `public_send'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:136:in `block (2 levels) in trigger_restricted_cops'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:160:in `with_cop_error_handling'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:135:in `block in trigger_restricted_cops'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:134:in `each'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:134:in `trigger_restricted_cops'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:70:in `on_send'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-ast-1.12.0/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-ast-1.12.0/lib/rubocop/ast/traversal.rb:137:in `each'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-ast-1.12.0/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-ast-1.12.0/lib/rubocop/ast/traversal.rb:20:in `walk'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/commissioner.rb:86:in `investigate'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/team.rb:155:in `investigate_partial'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cop/team.rb:83:in `investigate'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:309:in `inspect_file'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:253:in `block in do_inspection_loop'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:287:in `block in iterate_until_no_changes'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:280:in `loop'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:280:in `iterate_until_no_changes'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:249:in `do_inspection_loop'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:130:in `block in file_offenses'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:155:in `file_offense_cache'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:129:in `file_offenses'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:120:in `process_file'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:101:in `block in each_inspected_file'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:100:in `each'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:100:in `reduce'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:100:in `each_inspected_file'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:86:in `inspect_files'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/runner.rb:47:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/command.rb:11:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli/environment.rb:18:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli.rb:71:in `run_command'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli.rb:78:in `execute_runners'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/lib/rubocop/cli.rb:47:in `run'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/exe/rubocop:12:in `block in <top (required)>'
$HOME/.rvm/rubies/ruby-2.7.4/lib/ruby/2.7.0/benchmark.rb:308:in `realtime'
$HOME/.rvm/gems/ruby-2.7.4/gems/rubocop-1.22.0/exe/rubocop:12:in `<top (required)>'
$HOME/.rvm/gems/ruby-2.7.4/bin/rubocop:25:in `load'
$HOME/.rvm/gems/ruby-2.7.4/bin/rubocop:25:in `<main>'
$HOME/.rvm/gems/ruby-2.7.4/bin/ruby_executable_hooks:22:in `eval'
$HOME/.rvm/gems/ruby-2.7.4/bin/ruby_executable_hooks:22:in `<main>'
.

1 file inspected, no offenses detected

1 error occurred:
An error occurred while Lint/RequireRelativeSelfPath cop was inspecting $HOME/test/test.rb:4:0.
Errors are usually caused by RuboCop bugs.
Please, report your problems to RuboCop's issue tracker.
https://github.com/rubocop/rubocop/issues

Mention the following information in the issue report:
1.22.0 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 2.7.4 x86_64-darwin20)
Finished in 0.7015270000001692 seconds
```

## RuboCop version

Include the output of `rubocop -V` or `bundle exec rubocop -V` if using Bundler.
If you see extension cop versions (e.g. `rubocop-performance`, `rubocop-rspec`, and others)
output by `rubocop -V`, include them as well. Here's an example:

```
$ > rubocop -V
1.22.0 (using Parser 3.0.2.0, rubocop-ast 1.12.0, running on ruby 2.7.4 x86_64-darwin20)
  - rubocop-rspec 2.5.0
```

**Comment by koic** [comment]
This issue has been resolved by RuboCop 1.22.1. Can you update RuboCop to 1.22.1? Thank you.
